src/components/RegisterPage/indicatortable/indicatortablebody/indicatortablebody.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { OptsTu } from "../../../select_multi";
+import { RegisterNames } from "../../.";
+
+import TableBlock from "../tableblock/tableblock";
+
+export interface IndicatorTableBodyProps {
+  context: string;
+  tableType: "allRegistries" | "singleRegister";
+  optstu: OptsTu[] | [];
+  colspan: number;
+  registerNames: RegisterNames[];
+  unitNames: string[];
+  treatmentYear: number;
+  medicalFieldFilter: string[];
+  showLevelFilter: string;
+  blockTitle?: string[];
+}
+
+export const IndicatorTableBody: React.FC<IndicatorTableBodyProps> = (
+  props
+) => {
+  const {
+    context,
+    tableType,
+    optstu,
+    colspan,
+    registerNames,
+    unitNames,
+    treatmentYear,
+    medicalFieldFilter,
+    showLevelFilter,
+    blockTitle,
+  } = props;
+
+  const done: string[] = [];
+  let register_block = registerNames.map((register, i) => {
+    if (!done.includes(register.rname)) {
+      done.push(register.rname);
+
+      return (
+        <TableBlock
+          context={context}
+          tableType={tableType}
+          key={`${register.rname}`}
+          optstu={optstu}
+          registerName={register}
+          colspan={colspan}
+          unitNames={unitNames}
+          treatmentYear={treatmentYear}
+          medicalFieldFilter={medicalFieldFilter}
+          showLevelFilter={showLevelFilter}
+          blockTitle={blockTitle ? blockTitle[i] : undefined}
+        />
+      );
+    } else {
+      return null;
+    }
+  });
+
+  return <tbody>{register_block}</tbody>;
```

Comment:
Her kommer det veldig mange komponenter som fører til mange queries.
Kunne vært en ide å sjekke om komponenten var i viewport før man gjør querien.

Man kan.eks bruke [IntersectionObserver](http://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) sammen med `enabled` option på react-query eller kanskje react-window for å lage en lazy loading av kompontene her.

Reply:
Ja, høres fornuftig ut. Vi setter det på todo-listen og tar det i en egen PR.